src/components/UI/market/ProductForm.tsx

```diff
@@ -1,26 +1,30 @@
 import styled from 'styled-components';
 import axios from 'axios';
-import React, { useState } from 'react';
-import { useNavigate } from 'react-router-dom';
-import ProductInput from './ProductInput';
-import { postProduct } from '../../../api/market';
+import React, { useState, useEffect } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { postProduct, updateProduct } from '../../../api/market';
 import useToken from '../../../hooks/useToken';
+import ProductInput from './ProductInput';
+import { ProductDetail, ProductFormValues } from '../../../types';
+import { categories } from '../../../assets/data/categories';
 
-export interface ProductFormValues {
-  title: string;
-  price: number | string;
-  content: string;
+interface ProductFormProps {
+  productObj?: ProductDetail | undefined;
```

Comment:
? 에서 이미 undefined 를 내포하고 있기 때문에 불필요한 중복 코드임당
```suggestion
  productObj?: ProductDetail;
```